ts/src/base/errorHierarchy.ts

```diff
@@ -26,6 +26,7 @@ const errorHierarchy = {
                 'OrderImmediatelyFillable': {},
                 'OrderNotFillable': {},
                 'DuplicateOrderId': {},
+                'ContractSettled': {},
```

Comment:
@samgermain these (errorHierarchy and errors.ts) are the only files that need to be updated when adding a new error.

Reply:
The only changes that remain are those to errorHierarchy.ts and errors.ts (and okx, to actually use the new error)

I had made changes to python and php previously, but reverted those changes